gazelle/python/generate.go

```diff
@@ -485,6 +487,33 @@ func (py *Python) GenerateRules(args language.GenerateArgs) language.GenerateRes
 	return result
 }
 
+// getRulesWithInvalidSrcs checks existing Python rules in the BUILD file and return the rules with invalid srcs.
+func (py *Python) getRulesWithInvalidSrcs(args language.GenerateArgs) (invalidRules []*rule.Rule) {
+	if args.File == nil {
+		return
+	}
+	regularFiles := args.RegularFiles
+	regularFilesMap := make(map[string]struct{})
+	for _, file := range regularFiles {
+		regularFilesMap[file] = struct{}{}
+	}
+	for _, existingRule := range args.File.Rules {
+		if _, ok := py.Kinds()[existingRule.Kind()]; !ok {
+			continue
+		}
+		allInvalidSrcs := true
+		for _, src := range existingRule.AttrStrings("srcs") {
+			if _, ok := regularFilesMap[src]; ok {
```

Comment:
We also need to check:
1. whether it's in `args.GenFiles`
2. whether it's a target. If it starts with either `@`, `//` or `:`, we can assume it's a target. If not, we can see if it ends with ".py"
3. whether is a `glob`